compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -311,7 +311,8 @@ impl<'tcx> LayoutCx<'tcx, TyCtxt<'tcx>> {
         let dl = self.data_layout();
         let pack = repr.pack;
         if pack.is_some() && repr.align.is_some() {
-            bug!("struct cannot be packed and aligned");
+            self.tcx.sess.delay_span_bug(DUMMY_SP, "struct cannot be packed and aligned");
+            return Err(LayoutError::Unknown(ty));
```

Comment:
Is it correct for this and below to return `LayoutError::Unknown`? Is the idea that it doesn't matter because an error will have already been emitted and the `Unknown` will never be emitted to the user?

Reply:
Conceptually returning `LayoutError::Unknown` does seems correct to me. At the same time it wouldn't be exactly the kind of error we would like to emit, given that it is relatively uninformative.

The general idea is that the incorrect use of packed & aligned will diagnosed elsewhere, before we reach any code that is in position to assume that failure to compute layout is an error or a bug (like codegen does assume).